components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -331,11 +330,7 @@ public ServiceURLBuilder addFragmentParameter(String key, String value) {
 
     private String resolveTenantDomain() {
 
-        String tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
-        if (StringUtils.isBlank(tenantDomain)) {
-            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-        }
-        return tenantDomain;
+        return IdentityTenantUtil.getTenantDomainFromContext();
```

Comment:
In the non tenant qualified URL mode, shouldn't we get the tenantDomain from the context?

Reply:
In the non-tenant qualified mode, the ServiceURLBuilder does not obtain the tenant domain since it won't append that in the path.

Reply:
I don't think we should do this change now.
We should have it for non tenant mode. Appending the tenant to URL should be decided based on the config not from the tenant in context right

Reply:
+1